### ISSUE TITLE
Incorrect warning and fix for "goog.provide"

### ISSUE THREAD

**Issue by GoogleCodeExporter** [google/closure-linter#23]
```
What steps will reproduce the problem?

I have code that I want to be able to run without goog tools, thus I sniff for 
goog at the top of the file.

--- INPUT file.js ----
if (typeof goog != 'undefined' && typeof goog.provide == 'function') {
    goog.provide('MyLib.Base');
}
---------------------

gjslint complains: 
"Line 1, E:0141: goog.provide classes must be alphabetized.  The correct code 
is:
goog.provide('MochiKit.Base');
goog.provide('function');"

Running fixjsstyle outputs this... 
--- OUTPUT -----
if (typeof goog != 'undefined' && typeof goog.provide == 'MyLib.Base') {
    goog.provide('function');
}
----------------------

What version of the product are you using? On what operating system?
Latest build (Jan 17), Python 2.5.4, Windows 7 64bit

// Fredrik Blomqvist

```

Original issue reported on code.google.com by `fblomqvist@gmail.com` on 20 Feb 2011 at 1:23
